pkgs/build-support/rust/build-rust-package/default.nix

```
@@ -86,22 +75,31 @@ let
 
   sysroot = callPackage ./sysroot { } {
     inherit target shortTarget;
-    RUSTFLAGS = args.RUSTFLAGS or "";
-    originalCargoToml = src + /Cargo.toml; # profile info is later extracted
+    RUSTFLAGS = finalAttrs.RUSTFLAGS or "";
+    originalCargoToml = finalAttrs.src + /Cargo.toml; # profile info is later extracted
   };
 
 in
 
-# Tests don't currently work for `no_std`, and all custom sysroots are currently built without `std`.
-# See https://os.phil-opp.com/testing/ for more information.
-assert useSysroot -> !(args.doCheck or true);
 
-stdenv.mkDerivation ((removeAttrs args [ "depsExtraArgs" "cargoUpdateHook" "cargoLock" ]) // lib.optionalAttrs useSysroot {
-  RUSTFLAGS = "--sysroot ${sysroot} " + (args.RUSTFLAGS or "");
+previousAttrs:
+
+lib.optionalAttrs useSysroot {
+  RUSTFLAGS =
+    # Tests don't currently work for `no_std`, and all custom sysroots are currently built without `std`.
+    # See https://os.phil-opp.com/testing/ for more information.
```

Nix 2.4+ will show 1 previous line in the exception message.

```suggestion
    # Tests don't currently work for `no_std`, and all custom sysroots are currently built without `std`. See https://os.phil-opp.com/testing/ for more information.
```

alternatively, use `throwIfNot`

---

Thanks, changed it like this:
```nix
lib.optionalAttrs useSysroot {
  RUSTFLAGS = lib.throwIf (finalAttrs.doCheck or false) ''
    Tests don't currently work for `no_std`, and all custom sysroots are
    currently built without `std`. See https://os.phil-opp.com/testing/ for more
    information.
  '' ("--sysroot ${sysroot} " + (finalAttrs.RUSTFLAGS or ""));
}
```